### PR TITLE
New logos + Updated padding + fully customized buttons (both linked and not linked)

### PR DESCRIPTION
**Pull request by aidan-hubley**

### CODE REVIEW

**Review by 02NRA — COMMENTED** [review]
1. Unless you have a reason not to, we should import all of assets, not just one image at a time.
2. Alex mentioned index.js needing to be identical to login.js (I can send you his text). Is this no longer the case, or is this a problem in your code?

**Comment by dev-arekusandoru** [comment]
@02NRA @aidan-hubley i think my text was misunderstood, but no worries. here is why login and index were the same: when in development, the router looks for index.js from the start. because of this, it is helpful to code the new page on index.js so that you can see the changes there immediately without having to navigate to the page. so when i was developing the login page, i created it in index.js and then copied it to login.js at the end. i didn't see a point in deleting index all together so it made sense to me just to leave them both in there. that being said it is not a requirement for them to be identical

**Comment by dev-arekusandoru** [comment]
@aidan-hubley I had a peek at the code changes here and it seems like there is a disconnect between this branch and master. for some reason a lot of the changes that already exist on master are showing up as changes for this branch and other PRs you are submitting. are you pulling these branches off of the latest version of master? if so, it may be that they need to get the update from master which you can accomplish by merging master into your branch and then committing the changes. if you need more clarification i can definitely explain a bit further later today after work

**Comment by 02NRA** [comment]
Aidan, thanks for your in-person clarification on the image imports; no need to make changes there

**Comment by aidan-hubley** [comment]
> @02NRA @aidan-hubley i think my text was misunderstood, but no worries. here is why login and index were the same: when in development, the router looks for index.js from the start. because of this, it is helpful to code the new page on index.js so that you can see the changes there immediately without having to navigate to the page. so when i was developing the login page, i created it in index.js and then copied it to login.js at the end. i didn't see a point in deleting index all together so it made sense to me just to leave them both in there. that being said it is not a requirement for them to be identical

@dev-arekusandoru this makes a lot of sense to me. The next thing I plan to take out of our figma is the landing page, which I assume would remove the 'sandbox' aspect of index.js. Hence why I scrapped what we had previously and created my own pages to develop on. Thoughts?

**Comment by aidan-hubley** [comment]
> @aidan-hubley I had a peek at the code changes here and it seems like there is a disconnect between this branch and master. for some reason a lot of the changes that already exist on master are showing up as changes for this branch and other PRs you are submitting. are you pulling these branches off of the latest version of master? if so, it may be that they need to get the update from master which you can accomplish by merging master into your branch and then committing the changes. if you need more clarification i can definitely explain a bit further later today after work

@dev-arekusandoru This branch is and has been up to date with master. Aside from a handful of refactoring on previous code, there is little to no duplicate code changes from what master already has? If there was duplicate code, wouldn't it act as if it wasn't change in the first place?

**Comment by dev-arekusandoru** [comment]
@aidan-hubley yeah that sounds great. without routing it was impossible to test a page that wasn't index.js. however, since we have that functioning now, as soon as the landing page is completed it makes sense to leave that as index.js. as for the duplicate code, 

i really don't know what's going on here haha. per our slack message, i think it is fine if we go through with the merge, i just wanted to make sure so that we don't lose any of the previously written code.

File: components/Button.js [hunk 1]
```diff
@@ -47,16 +56,24 @@ export function Button({
 			<StyledLink href={href ? href : null} asChild>
 				<StyledTouchableHighlight
 					activeOpacity={0.6}
-					underlayColor="#DDDDDD"
+					// underlayColor="#DDDDDD"
```
Comment (dev-arekusandoru) [inline components/Button.js]: same as above

File: components/Button.js [hunk 2]
```diff
@@ -19,24 +20,32 @@ export function Button({
...
 
 	if (!href) {
 		return (
 			<StyledTouchableHighlight
 				activeOpacity={0.6}
-				underlayColor="#DDDDDD"
-				className={`flex h-[50px] items-center justify-center rounded-full bg-offwhite border border-[${borderClr}]
-				${width ? width : "w-11/12"}
-			`}
+				// underlayColor="#DDDDDD"
```
Comment (dev-arekusandoru) [inline components/Button.js]: from my understanding this underlayColor is needed to have the effect when pressing on the button. not entirely sure if that is right but if it doesn't do that then we can probably just delete it

File: app/login.js [hunk 3]
```diff
@@ -28,76 +29,86 @@ export default function Login() {
 		<>
 			<StyledSafeArea
 				className="bg-offblack"
-				style={{ flex: 1 /*  backgroundColor: "#5946B2"  */ }}
+				style={{ flex: 1, backgroundColor: "#5946B2" }}
```
Comment (dev-arekusandoru) [inline app/login.js:32]: should convert these to nativewind classnames

File: app/login.js [hunk 4]
```diff
@@ -28,76 +29,86 @@ export default function Login() {
...
-									source={require("../assets/PCLogo.png")}
-									resizeMode="contain"
-								/>
+				<KeyboardAwareScrollView>
+					<TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+						<>
+
+						<StyledView className="flex flex-col pb-5 px-[15px] w-screen">
+							<StyledView className="w-full flex flex-col items-center mb-2 pt-[19%]">
+								{/* <StyledText className="text-offwhite text-center text-5xl font-bold">
```
Comment (dev-arekusandoru) [inline app/login.js]: We can delete this if we don't think it's necessary. i personally dont care, so maybe just check with everyone real quick

File: app/register.js [hunk 5]
```diff
@@ -33,20 +33,20 @@ export default function Login() {
 		<>
 			<StyledSafeArea
 				className="bg-offblack"
-				style={{ flex: 1 /*  backgroundColor: "#5946B2"  */ }}
+				style={{ flex: 1, backgroundColor: "#5946B2" }}
```
Comment (dev-arekusandoru) [inline app/register.js:36]: like above, this should be converted to nativewind

File: app/register.js [hunk 6]
```diff
@@ -33,20 +33,20 @@ export default function Login() {
...
 						<>
-							<StyledView className="flex flex-col pt-10 pb-5 px-[15px] items-center justify-start w-screen ">
-								<StyledView className="w-full flex flex-col items-center justify-center mb-2">
-									<StyledText className="text-offwhite text-center text-5xl font-bold">
-										Welcome to Prayer Circle!!
-									</StyledText>
-									<StyledView className="w-[40%] aspect-square my-2">
+							<StyledView className="flex flex-col pb-5 px-[15px] w-screen">
+								<StyledView className="w-full flex flex-col items-center mb-2">
+									{/* <StyledText className="text-offwhite text-center text-5xl font-bold">
```
Comment (dev-arekusandoru) [inline app/register.js]: Also can be deleted if thats what we decide

File: app/register.js [hunk 7]
```diff
@@ -33,20 +33,20 @@ export default function Login() {
...
+										Welcome to Prayer Circle!
+									</StyledText> */}
+									<StyledView className="w-[89%] aspect-square mt-[15%] mb-[10%]">
```
Comment (dev-arekusandoru) [inline app/register.js:43]: not sure percentages are the best for margin. if we go with this, we need to test on different screen sizes to make sure everything looks good. I would suggest using static values, even if we use our own values(not the prebuilt ones): `m-[10px]`

File: components/Button.js [hunk 8]
```diff
@@ -19,24 +20,32 @@ export function Button({
...
+				className={`flex items-center justify-center rounded-full 
+					${height ? height : "h-[50px]"} 
+					${width ? width : "w-11/12"}
+				`}
+				style={{
```
Comment (dev-arekusandoru) [inline components/Button.js:37]: here, the bg color can be added to the nativewind styles

File: components/Button.js [hunk 9]
```diff
@@ -47,16 +56,24 @@ export function Button({
...
+					// underlayColor="#DDDDDD"
 					onPress={() => press()}
-					className={`flex h-[50px] items-center justify-center rounded-full bg-offwhite border border-[${borderClr}]
-				${width ? width : "w-11/12"}
-			`}
+					className={`flex items-center justify-center rounded-full 
+						${height ? height : "h-[50px]"} 
+						${width ? width : "w-11/12"}
+					`}
+					style={{
```
Comment (dev-arekusandoru) [inline components/Button.js:67]: same as above

File: components/Button.js [hunk 10]
```diff
@@ -47,16 +56,24 @@ export function Button({
...
 				>
 					<StyledText
-						className={`font-bold text-[${txtColor}] ${
-							textSize ? textSize : "text-[20px]"
-						} ${textStyles}`}
+						className={`font-bold 
+							${textSize ? textSize : "text-[20px]"}
+							${textStyles}
+						`}
+						style={{ color: txtColor }}
```
Comment (dev-arekusandoru) [inline components/Button.js:78]: should be added as in nativewind

**Review by dev-arekusandoru — DISMISSED** [review]
a couple things i noticed. nothing crucial to functionality, mainly just some clean up and maintaining standards. especially when it comes to styles, i think it is super important to go with either nativewind or default styles on an element, unless it is absolutely necessary to mix the two

**Comment by aidan-hubley** [comment]
@dev-arekusandoru overall I agree with your suggestions but I want to mention a couple of things

When developing these, I used a variety of screen sizes to ensure the percentage based padding/margin was appropriate. 

The native wind styles are great, but I had to add the vanilla styling to the components to get parameters to function. TL;DR the nativewind class name wasn't taking my component parameters. While the vanilla styling does take my parameters. Does it matter if we use the best of both worlds? Is there any benefit to pure native wind?

**Comment by dev-arekusandoru** [comment]
@aidan-hubley if it works in testing then that's great. i just know that as a general standard with styling percentage is not the best way to go. i would suggest using a static margin value and then set a max-width so that it doesn't scale to big on big screen sizes. let me know if that makes sense.

as for the nativewind styles, i think there is a way to get them to work with parameters(see my existing code). here is an example of using your txtColor parameter in nativewind that will work: `className=`text-[${txtColor}]`. try that out and let me know if it fails. I wouldn't say it is the end of the world if we use a mixture of both styles but I would argue it to be a last resort. we chose nativewind as the standard and so i think it should be used to its maximum as such. of course we will probably run into cases where it just doesn't make the cut and in those instances i am more than happy to use a combination of the two. for the styles in this PR, however, i don't think there is anything here that nativewind can't or shouldn't handle.

File: components/Button.js [hunk 11]
```diff
@@ -19,24 +20,32 @@ export function Button({
...
 			<StyledTouchableHighlight
 				activeOpacity={0.6}
-				underlayColor="#DDDDDD"
-				className={`flex h-[50px] items-center justify-center rounded-full bg-offwhite border border-[${borderClr}]
-				${width ? width : "w-11/12"}
-			`}
+				// underlayColor="#DDDDDD"
+				className={`flex items-center justify-center rounded-full 
+					${height ? height : "h-[50px]"} 
+					${width ? width : "w-11/12"}
```
Comment (dev-arekusandoru) [inline components/Button.js]:
```suggestion
					${width ? width : "w-11/12"} border bg-[${backgroundColor}] border-[${borderColor}]
```
here is how I would convert the vanilla styles to nativewind

File: components/Button.js [hunk 12]
```diff
@@ -19,24 +20,32 @@ export function Button({
...
+					}}
 				onPress={() => press()}
 			>
 				<StyledText
-					className={`font-bold text-[${txtColor}] ${
-						textSize ? textSize : "text-[20px]"
-					} ${textStyles}`}
+					className={`font-bold 
+						${textSize ? textSize : "text-[20px]"}
+						${textStyles}
```
Comment (dev-arekusandoru) [inline components/Button.js]:
```suggestion
						${textStyles} text-[${txtColor}]
```
conversion for text color

**Comment by 02NRA** [comment]
@aidan-hubley let me know when you have made the changes that you would like to make based on @dev-arekusandoru 's suggestions, and we'll see what we can do about the merge conflict here.

File: app/login.js (hunk 3 above)
Comment (aidan-hubley) [reply app/login.js:32]: If you can, feel free to. @dev-arekusandoru

File: app/register.js (hunk 5 above)
Comment (aidan-hubley) [reply app/register.js:36]: If you can, feel free to. @dev-arekusandoru

File: components/Button.js (hunk 8 above)
Comment (aidan-hubley) [reply components/Button.js:37]: If you can, feel free to. @dev-arekusandoru

File: components/Button.js (hunk 9 above)
Comment (aidan-hubley) [reply components/Button.js:67]: If you can, feel free to. @dev-arekusandoru

File: components/Button.js (hunk 10 above)
Comment (aidan-hubley) [reply components/Button.js:78]: If you can, feel free to. @dev-arekusandoru

**Comment by aidan-hubley** [comment]
@dev-arekusandoru tried some of your suggestions. They don't throw errors but also don't work. Specifically the nativewind background colors. Feel free to attempt and remidy this but from my perspective we have bigger fish to fry and a dull nativewind / vanilla styles is acceptable

**Review by 02NRA — APPROVED** [review]
Thanks for all your hard work and communication on branch compatibility on this one.

**Pull request merged**